### PR TITLE
chore: Add github pull request template

### PR DESCRIPTION
**Pull request by quartzmo**
Based on: [synthtool/synthtool/gcp/templates/node_library/.github/PULL_REQUEST_TEMPLATE.md](https://github.com/googleapis/synthtool/blob/c91c3560bfba5a532a067187d9dc28d3499d15df/synthtool/gcp/templates/node_library/.github/PULL_REQUEST_TEMPLATE.md)

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,8 @@
+Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
+
+- [ ] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/{{metadata['repo']['name']}}/issues) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea.
```
Comment (dazuma) [inline .github/PULL_REQUEST_TEMPLATE.md:3]: Just hard-code `google-cloud-ruby` as the repo name.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -0,0 +1,8 @@
...
+- [ ] Ensure the tests and linter pass.
```
Comment (dazuma) [inline .github/PULL_REQUEST_TEMPLATE.md:4]: Give a brief note on how to do so (e.g. `bundle exec rake ci`) or link to instructions elsewhere (e.g. in `.github/CONTRIBUTING.md`)

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 3]
```diff
@@ -0,0 +1,8 @@
...
+- [ ] Code coverage does not decrease (if any source code was changed.)
```
Comment (dazuma) [inline .github/PULL_REQUEST_TEMPLATE.md:5]: Again, link to how to determine this. For us, though, I'm not sure we pay that much attention to this, nor is our tooling really in good shape to track/report this, so we might want to omit it.

Comment (quartzmo) [reply .github/PULL_REQUEST_TEMPLATE.md]: Removed.

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 2 above)
Comment (quartzmo) [reply .github/PULL_REQUEST_TEMPLATE.md]: Done.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 4]
```diff
@@ -0,0 +1,7 @@
+Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
+
+- [ ] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/google-cloud-ruby/issues) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea.
+- [ ] Follow the instructions in [CONTRIBUTING](CONTRIBUTING.md). Most importantly, ensure the tests and linter pass by running `bundle exec rake ci` in the gem subdirectory.
```
Comment (quartzmo) [inline .github/PULL_REQUEST_TEMPLATE.md:4]: @dazuma Looking at [CONTRIBUTING.md#tests](https://github.com/googleapis/google-cloud-ruby/blob/master/.github/CONTRIBUTING.md#tests) just now, I noticed that we omit `bundle exec` from the `rake` command examples. I think this dates from long ago when we were much more agnostic about using Bundler. Should I update those usages in this PR?

Comment (quartzmo) [reply .github/PULL_REQUEST_TEMPLATE.md:4]: Done.

**Pull request merged**